File: code-awesome/java-quarkus-vue/todo/frontend/src/pages/create-todo-list.js
// create-todo-list.js

import { createApp } from 'vue';
import App from './App.vue';

const app = createApp(App);

// Inject data from server - example
const serverData = {
  'title': 'My Awesome List',
  'description': 'This is where I track everything!'
};

app.provide('serverData', serverData);

// Method to create a new todo list
function createTodoList(title, description) {
  const newList = {
    title: title,
    description: description,
    items: [] // Empty list to hold todo items
  };

  //Store the list in the Vue instance
  app.config.globalProperties.currentTodoList = newList;
}

// Component for the create todo list form
app.mount('#app')

File: code-awesome/java-quarkus-angular/todo/backend/src/pages/create-todo-list.java
package pages;

import com.google.inject.Inject;
import com.google.inject.Provider;

import java.util.HashMap;
import java.util.Map;

public class CreateTodoList {

    private final Provider<TodoListService> todoListServiceProvider;

    @Inject
    public CreateTodoList(Provider<TodoListService> todoListServiceProvider) {
        this.todoListServiceProvider = todoListServiceProvider;
    }

    public String create(String title, String description) {
        TodoListService todoListService = todoListServiceProvider.get();
        String newTodoListId = todoListService.createTodoList(title, description);

        Map<String, Object> data = new HashMap<>();
        data.put("id", newTodoListId);
        data.put("title", title);
        data.put("description", description);

        return "{\"id\":\"" + newTodoListId + "\", \"title\":\"" + title + "\", \"description\":\"" + description + "\" }";
    }
}